packages/grid/_modules_/grid/components/containers/GridMainContainer.tsx

```diff
@@ -26,6 +26,7 @@ const GridMainContainerRoot = styled('div', {
   flexGrow: 1,
   display: 'flex',
   flexDirection: 'column',
+  overflow: 'hidden',
```

Comment:
Fixes the missing right border in https://deploy-preview-2946--material-ui-x.netlify.app/components/data-grid/columns/#pinning-the-checkbox-selection-column

<img width="80" src="https://user-images.githubusercontent.com/42154031/142016891-26b8dbed-fa57-46a7-b18d-dc6d46e0279f.png" alt="image" style="max-width: 100%;">